backend/src/modules/match-module/core/domain/Match.ts

```diff
@@ -43,10 +48,46 @@ export function startMatch(
   };
 }
 
+export function endMatch(
+  state: Match | undefined,
+  command: { matchId: MatchId; winner: MatchSideId },
+  currentTime: Date,
+): DomainCommandResult<Match> {
+  if (!state?.matchId) {
+    throw new Error("Cannot end match that hasn't started.");
+  }
+  if (!state.firstMatchSideId.equals(command.winner) && !state.secondMatchSideId.equals(command.winner)) {
```

Comment:
Super, że o tym pamiętałeś :) Ale może zamiast wyciągać wszystko na zewnątrz to np. w Match można zrobić metodę w stylu `isParticipating(player: MatchSideId)` :) ?